metaseq/launcher/tombyard.py

```diff
@@ -0,0 +1,126 @@
+        "REVOKED",
+        "SIGNALING",
+        "SPECIAL_EXIT",
+        "STAGE_OUT",
+        "STOPPED",
+        "SUSPENDED",
+        "TIMEOUT",
+    ]
+
+    tombstone_detected = False
+    period = period_before_tombstone_detected
+
+    while True:
+        sacct_result = os.popen(f"squeue -j {job_id} -O State -h ").read()
+        status = sacct_result.strip()
+        if tombstone_detected:
+            print(f".. scanceling the job and its current squeue.state is {status}")
+        if status not in job_state_codes:
+            print(f"Done scanceling the job. Its squeue.state now is: {status}")
+            return
+        if not tombstone_detected:
+            for tombstone_name in dirstones["scancel"]:
+                if os.path.exists(tombstone_name):
+                    print(
+                        f"""tombstones_procedure has detected file {tombstone_name}.
+                          scancel {job_id} will be called every {period_after_tombstone_detected}
+                          until the job is dead"""
+                    )
+                    tombstone_detected = True
+                    period = period_after_tombstone_detected
+            for tombstone_name in dirstones["requeuehold"]:
+                if os.path.exists(tombstone_name):
+                    print(
+                        f"""tombstones_procedure has detected file {tombstone_name}.
+                          scontrol requeuehold {job_id} will be called once.
+                         remove the file {tombstone_name} within the next {period_before_tombstone_detected}
+                         for it not to trigger the same command again"""
+                    )
+                    _ = os.popen(
+                        f"""
+                        scontrol requeuehold {job_id}
+                         """
+                    ).read()
+            for tombstone_name in dirstones["requeuehold"]:
+                if os.path.exists(tombstone_name):
+                    print(
+                        f"""tombstones_procedure has detected file {tombstone_name}.
+                          scontrol release {job_id} will be called once.
+                         remove the file {tombstone_name} within the next {period_before_tombstone_detected}
+                         for it not to trigger the same command again"""
+                    )
+                    _ = os.popen(
+                        f"""
+                        scontrol release {job_id}
+                         """
+                    ).read()
+        if tombstone_detected:
+            _ = os.popen(
+                f"""
+                    scancel {job_id}
+                    """
+            ).read()
+        time.sleep(period.total_seconds())
+
+
+def tombstones(job_id, period=datetime.timedelta(seconds=60), dirstones=None):
+    if dirstones is None:
+        dirstones = {"scancel": [], "requeuehold": [], "release": []}
+        azure_base = "/shared/home"
```

Comment:
we're hardcoding here i believe

Reply:
changed it to import the correct base path from constants.py